go/ql/src/experimental/frameworks/DecompressionBombs.qll

```diff
@@ -0,0 +1,733 @@
+import go
+
+module DecompressionBombs {
+  class FlowState = DataFlow::FlowState;
```

Comment:
If I understand correctly, you're looking for flow from an `UntrustedFlowSource` (call this A) to a function that makes one of the zip readers that you are interested in (call this B) and the on to a function which tries to read from the zip reader (call this C). And you're using a flow state to make sure that we go through one of the B nodes, and don't just go from A to C some other way.

That is a good use for flow states. But I don't see why you need so many flow states. You can just have two (A and B, or "untrusted flow source" and "untrusted zip reader", or whatever).

Reply:
You do not have to change it. My preference is for just two flow states, but if you prefer it this way then you can leave it like this. If you leave it like this perhaps you could add a comment explaining what I summarised in my last comment, so future people reading it will understand what role the flow state plays.